### PR TITLE
update session

### PR DESCRIPTION
**Pull request by Maxgis**
1. delete not necssary code

1.  fix bug, one session can be overwrited by another request at the same time

### CODE REVIEW

File: controller.go [hunk 1]
```diff
@@ -499,46 +495,28 @@ func (c *Controller) SaveToFile(fromfile, tofile string) error {
...
-	return c.CruSession
-}
-
 // SetSession puts value into session.
 func (c *Controller) SetSession(name interface{}, value interface{}) {
-	if c.CruSession == nil {
-		c.StartSession()
-	}
-	c.CruSession.Set(name, value)
+	c.Ctx.Input.CruSession.Set(name, value)
```
Comment (astaxie) [inline controller.go]: if c.Ctx.Input.CruSession is nil, it will cause panic here and above

**Comment by astaxie** [comment]
@Maxgis any updated on it?

**Comment by Maxgis** [comment]
Sorry, I will fixed it today.

**Comment by Maxgis** [comment]
Only session is off , c.Ctx.Input.CruSession  will be nil. So I think it is not a bug.

**Comment by astaxie** [comment]
@Maxgis no. We should never panic inside beego. As if the session is turn off, but you still call this function it should never panic

**Comment by astaxie** [comment]
I will close this PR as you didn't have any update on it

**Pull request closed without merging**